htdocs/product/card.php

```diff
@@ -475,6 +478,9 @@
 				$object->status                 = GETPOST('statut', 'int');
 				$object->status_buy             = GETPOST('statut_buy', 'int');
 				$object->status_batch = GETPOST('status_batch', 'aZ09');
+				if ($object->status_batch !== 0) {
+					$object->batch_mask = GETPOST('batch_mask', 'alpha');
```

Comment:
Same. Just use  =  with no condition.
Condition should appears when we will need to know if we must use or not batch_mask.